etl/steps/data/meadow/nasa/2023-03-06/ozone_hole_area.py

```diff
@@ -0,0 +1,136 @@
+"""Load a snapshot and create a meadow dataset."""
+
+import re
+from pathlib import Path
+from typing import List, Union
+
+import pandas as pd
+from owid.catalog import Table
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+from etl.snapshot import Snapshot
+
+# Initialize logger.
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("ozone_hole_area.start")
+
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap_1: Snapshot = paths.load_dependency("ozone_hole_area_p1.txt")
+    snap_2: Snapshot = paths.load_dependency("ozone_hole_area_p2.txt")
+
+    # Load data from snapshot.
+    df = build_df(snap_1, snap_2)
+
+    #
+    # Process data.
+    #
+    # Create a new table and ensure all columns are snake-case.
+    tb = Table(df, short_name=paths.short_name, underscore=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new meadow dataset with the same metadata as the snapshot.
+    ds_meadow = create_dataset(dest_dir, tables=[tb], default_metadata=snap_1.metadata)
+
+    # Save changes in the new garden dataset.
+    ds_meadow.save()
+
+    log.info("ozone_hole_area.end")
+
+
+def build_df_p1(snap: Snapshot) -> pd.DataFrame:
+    """Read raw data from source 1 and build a dataframe."""
+    log.info("ozone_hole_area: loading data from source 1")
+    # Read data
+    text = _read_txt_without_comments(snap.path)
+    # Sanity checks
+    assert text[0] == "         Ozone Hole Area       Minimum Ozone\n", "Check data file header, might have changed!"
```

Comment:
These assertions are likely to fail in the next update, if they change just a single space. You could relax them a bit using regex, for example:
```
assert re.match("\s+Ozone Hole Area\s+Minimum Ozone\n", text[0]), "Data file header might have changed."
```

Reply:
Nice one!